pkg/cmd/provisioning/azure/create_oidc_issuer.go

```diff
@@ -0,0 +1,588 @@
+		storageAccountName,
+		containerName,
+		armstorage.BlobContainer{
+			ContainerProperties: &armstorage.ContainerProperties{
+				PublicAccess: to.Ptr(armstorage.PublicAccessContainer),
+				// Note there is no Tags parameter within ContainerProperties.
+			},
+		},
+		&armstorage.BlobContainersClientCreateOptions{},
+	)
+	if err != nil {
+		return nil, err
+	}
+	log.Printf("Created blob container %s", *createBlobContainerResp.BlobContainer.ID)
+	return &createBlobContainerResp.BlobContainer, nil
+}
+
+// uploadOIDCDocuments generates and uploads the OIDC discovery document (.well-known/openid-configuration) and the JSON web key set (jwks.json)
+// to the blob container
+func uploadOIDCDocuments(client *azureclients.AzureClientWrapper, storageAccountName, storageAccountKey, publicKeyFilepath, blobContainerName, targetDir string, dryRun bool) (string, error) {
+	blobContainerURL := fmt.Sprintf("https://%s.blob.core.windows.net/%s", storageAccountName, blobContainerName)
+
+	oidcDiscoveryDocumentData := []byte(fmt.Sprintf(openidConfigurationTemplate, blobContainerURL, blobContainerURL))
+	oidcDiscoveryDocumentFullPath := filepath.Join(targetDir, openidConfigurationFileName)
+	err := os.WriteFile(oidcDiscoveryDocumentFullPath, oidcDiscoveryDocumentData, fs.FileMode(fileMode))
+	if err != nil {
+		return blobContainerURL, err
+	}
+	log.Printf("Saved OIDC discovery document at path %s", oidcDiscoveryDocumentFullPath)
+
+	jwksFullPath := filepath.Join(targetDir, jwksFileName)
+	jwksData, err := provisioning.BuildJsonWebKeySet(publicKeyFilepath)
+	if err != nil {
+		return blobContainerURL, err
+	}
+	err = os.WriteFile(jwksFullPath, jwksData, fs.FileMode(fileMode))
+	if err != nil {
+		return blobContainerURL, err
+	}
+	log.Printf("Saved JSON web key set at path %s", jwksFullPath)
+
+	// Return before uploading documents if doing a dry run
+	if dryRun {
+		return blobContainerURL, nil
+	}
+
+	sharedKeyCredential, err := azblob.NewSharedKeyCredential(storageAccountName, storageAccountKey)
+	if err != nil {
+		log.Fatal(err)
+		return blobContainerURL, err
+	}
+	// NOTE: It is not possible to instantiate an azureclients.AzureClientWrapper with this client
+	// because the storage key isn't known when we instantiate the other clients because the storage
+	// account doesn't or may not exist yet.
+	//
+	// client.BlobSharedKeyClient is previously set in tests for mocking so only create a real client
+	// if client.BlobSharedKeyClient is nil.
+	if client.BlobSharedKeyClient == nil {
+		client.BlobSharedKeyClient, err = azureclients.NewAZBlobClientWithSharedKeyCredential(blobContainerURL, sharedKeyCredential, nil)
+		if err != nil {
+			return blobContainerURL, err
+		}
+	}
+
+	_, err = client.BlobSharedKeyClient.UploadBuffer(
+		context.TODO(),
+		"",
+		filepath.Join(".well-known", openidConfigurationFileName),
+		oidcDiscoveryDocumentData,
+		// Note: Blobs can be tagged in this options object but the storage account they reside in is tagged.
```

Comment:
Code review: tag the files (blobs) even though we can't tag the blob container. Also squirrely comment here because its the blob container that cannot be tagged.